### ISSUE TITLE
Support Mutating Manifest Container Image/Registry

### ISSUE THREAD

**Issue by seizadi** [argoproj/argo-cd#7627]
# Summary
We have several changes we make to the helm chart on the cluster to deal with container registry and image issues.
We make these changes on the cluster using [mutating hooks](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#mutatingadmissionwebhook). Specifically we use the following:
- [tugger](https://github.com/jainishshah17/tugger)
- [registry-creds](https://github.com/alexellis/registry-creds)

After post-rendering stage  mutations could be made before the rendered manifest is applied to the cluster. These changes also would be part of the logic to compare deployed manifest with Git Repo so that mutations are made before comparison.

# Motivation
- We have use case where we are running our own private repo and want to use third party charts where the container registry is not in the values file to override. The standard helm chart template when created has the registry and image combined:
```yaml
image:
  repository: nginx
  pullPolicy: IfNotPresent
  # Overrides the image tag whose default is the chart appVersion.
  tag: ""
```

- The other problem that the registry-cred webhook address is making sure that the ImagePullSecrets are configured in the namespace. This is needed to make our private registry work with third party charts. In clusters that are not in production and have access to dockerhub we started to experience access issues due to [rate limiting](https://www.docker.com/pricing) changes that took effect recently. So we are configuring an account for dockerhub so that we are not rate limited on access.

# Proposal
If these changes are of interest we could evaluate submitting a PR for the changes.

**Comment by jannfis** [comment]
Could that be handled with something as suggested in #7623?

**Comment by seizadi** [comment]
Yes it would, I add one comment on requirement. We can close this Issue if we move that one forward.
